sentry/src/main/java/io/sentry/SentryItemType.java

```diff
@@ -13,6 +13,18 @@
 
   private final String itemType;
 
+  public static SentryItemType resolve(Object item) {
+    if (item instanceof SentryEvent) {
+      return Event;
+    } else if (item instanceof SentryTransaction) {
+      return Transaction;
+    } else if (item instanceof Session) {
+      return Session;
+    } else {
+      return Attachment;
```

Comment:
Attachment should have type `json` (which isn't visible here but it's the state the envelope item ends up in)